external_search.py

```diff
@@ -33,114 +42,158 @@ def __init__(self, url=None, works_index=None):
                 if not works_index:
                     works_index = integration.get(
                         Configuration.ELASTICSEARCH_INDEX_KEY
-                    ) or None
+                    )
 
                 if not url:
                     if not integration:
                         return
                     url = integration[Configuration.URL]
 
-            use_ssl = url and url.startswith('https://')
+            if not url:
+                raise Exception("Cannot connect to Elasticsearch cluster.")
+            use_ssl = url.startswith('https://')
             self.log.info(
                 "Connecting to index %s in Elasticsearch cluster at %s", 
                 works_index, url
             )
             ExternalSearchIndex.__client = Elasticsearch(
                 url, use_ssl=use_ssl, timeout=20, maxsize=25
             )
-            ExternalSearchIndex.__client.works_index = works_index
-            if not url:
-                raise Exception("Cannot connect to Elasticsearch cluster.")
 
-        self.works_index = self.__client.works_index
         self.indices = self.__client.indices
         self.search = self.__client.search
         self.index = self.__client.index
         self.delete = self.__client.delete
         self.exists = self.__client.exists
+
+        # Sets self.works_index and self.works_alias values.
+        # Document upload runs against the works_index.
+        # Search queries run against works_alias.
+        self.works_index = None
+        self.works_alias = None
+        self.set_works_index_and_alias(works_index)
+
         def bulk(docs, **kwargs):
             return elasticsearch_bulk(self.__client, docs, **kwargs)
         self.bulk = bulk
-            
+
+    def set_works_index_and_alias(self, current_alias):
+        """Finds or creates the index based on provided configuration"""
+        index_details = self.indices.get_alias(name=current_alias, ignore=[404])
```

Comment:
It returns a dict of the index and it's alias information, like`{"circulation-works-v2" : { "aliases" : { "circulation-works-current" : {}, "other-alias" : {}}}}`. If you use the `ignore` keyword, it will return something like `{ "status" : 404, "error" : "Not found" }` if the alias doesn't exist.